source/jormungandr/jormungandr/scenarios/journey_filter.py

```diff
@@ -525,6 +525,26 @@ def apply_final_journey_filters(response_list, instance, request):
     _filter_too_much_connections(journeys, instance, request)
 
 
+def filter_detailed_journeys(responses, request):
+    journey_generator = get_qualified_journeys
+    if request.get('debug', False):
+        journey_generator = get_all_journeys
+
+    journeys = journey_generator(responses)
+
+    min_bike = request.get('_min_bike', None)
+    min_car = request.get('_min_car', None)
+    orig_modes = request.get('origin_mode', [])
+    dest_modes = request.get('destination_mode', [])
+
+    too_heavy_journey_filter = FilterTooShortHeavyJourneys(
+        min_bike=min_bike, min_car=min_car, orig_modes=orig_modes, dest_modes=dest_modes
+    )
```

Comment:
Should be removed from `filter_journeys`, no?

Reply:
It can also remain here. It'll optimize properly when it's Valhalla

Reply:
I vote letting it in `filter_journeys`, early one-by-one filter is probably better in perf considering mean and worst case.
I'm just wondering if we can mutualize code more (param retrieving + journey generator)